File: regexlib/2021-5-15/javascript_test_file/regexlib_5271.js
/* 5271
 * (\s*\S*){2}(ipsum)(\S*\s*){2}
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"\t"*512+"!_1_NQ"
 */
var REGEX = /(\s*\S*){2}(ipsum)(\S*\s*){2}/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '\t'.repeat(i*1) + '!_1_NQ'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/python_re2_test_file/regexlib_5271.py
# 5271
# (\s*\S*){2}(ipsum)(\S*\s*){2}
# EXPONENT
# nums:5
# EXPONENT AttackString:""+" "*512+"!_1_NQ"

import re2 as re
from time import perf_counter

regex = """(\s*\S*){2}(ipsum)(\S*\s*){2}"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + " " * i * 1 + "!_1_NQ"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")